src/community/infix-abstract.ts

```diff
@@ -0,0 +1,79 @@
+import { Notation } from "../notation";
+import Decimal, { DecimalSource } from "break_infinity.js";
+import { fixMantissaOverflow } from '../utils';
+
+export abstract class AbstractInfixNotation extends Notation {
+  public get name(): string {
+    return "Abstract Infix";
+  }
+  
+  protected groupDigits = 3;
+  protected canHandleZeroExponent = true;
+  
+  protected abstract formatMantissa(digit: number): string;
+  protected abstract formatExponent(digit: number): string;
+  
+  private nextSeparatorExponent(e: number) {
+    // Get the next exponent (going down, so the highest exponent lower than e)
+    // such that there is a separator at that exponent.
+    let modulus = (0 <= e && e < this.groupDigits) ? 3 : this.groupDigits;
+    return e - (e % modulus + modulus) % modulus;
+  }
+  
+  public formatDecimal(value: Decimal, places: number): string {
+    return this.formatInfix(value, places);
+  }
+  
+  public format(value: DecimalSource, places=0, placesUnder1000=0): string {
```

Comment:
Codacy found an issue: [Declaration of public instance method not allowed after declaration of private instance method. Instead, this should come after protected instance fields.](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6218257)